drivers/serial/uart_nrfx_uarte.c

```diff
@@ -170,32 +204,46 @@ static inline NRF_UARTE_Type *get_uarte_instance(const struct device *dev)
 static void uarte_nrfx_isr_int(void *arg)
 {
 	const struct device *dev = arg;
-	struct uarte_nrfx_data *data = get_dev_data(dev);
 	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
 
-	if (data->int_driven->disable_tx_irq &&
-	    nrf_uarte_event_check(uarte, NRF_UARTE_EVENT_ENDTX)) {
-		nrf_uarte_int_disable(uarte, NRF_UARTE_INT_ENDTX_MASK);
-
-		/* If there is nothing to send, driver will save an energy
-		 * when TX is stopped.
-		 */
-		nrf_uarte_task_trigger(uarte, NRF_UARTE_TASK_STOPTX);
+	/* If interrupt driven and asynchronous APIs are disabled then UART
+	 * interrupt is still called to stop TX. Unless it is done using PPI.
+	 */
+	if (nrf_uarte_int_enable_check(uarte, NRF_UARTE_INT_ENDTX_MASK) &&
+		nrf_uarte_event_check(uarte, NRF_UARTE_EVENT_ENDTX)) {
+		endtx_isr(dev);
```

Comment:
This should be done only `if (!get_dev_config(dev)->ppi_endtx)`. Also in `uarte_nrfx_isr_async()`, so perhaps it would be then good to move all those checks to `endtx_isr`?

Reply:
but  when ppi is used then ENDTX interrupt is never enabled so `nrf_uarte_int_enable_check(uarte, NRF_UARTE_INT_ENDTX_MASK)` is equivalent to `!get_dev_config(dev)->ppi_endtx`

Reply:
Right, I missed that.